### ISSUE TITLE
Problem with settings files

### ISSUE THREAD

**Issue by Lochabar** [chummer5a/chummer5a#3532]
**Describe the bug**
I have two settings file, "default" and "missions". The "default" file basically has all options set. Whereas "missions" is specifically for the SR Missions options.

I've been working on a Custom Data Dictionary for Neo-Tokyo. When I selected the "missions" settings file I realized Chummer is still showing the "default" settings.

**To Reproduce**
1. Create a second settings file with different Custom Data Directories selected 
2. Go to 'Tools' (no characters open) and select 'Options'
3. Click on 'Optional Rules' and notice the selected directories
4. Select the alternative Settings File
5. See that the "default" directories are still shown not the alternative ones.

**Expected behavior**
The selected settings file is actually shown and saved with changes and remains separate from any other settings file.

**Screenshots**
"default" Optional Rules
![default](https://user-images.githubusercontent.com/17562417/55241294-c764d180-5208-11e9-90d4-f60eddb8e246.png)

this corresponds to the settings file
![default directories](https://user-images.githubusercontent.com/17562417/55241325-d3e92a00-5208-11e9-81f4-0996fe09fc6f.png)

"missions" Optional Rules
![missions](https://user-images.githubusercontent.com/17562417/55241355-e2cfdc80-5208-11e9-8b8e-a4e9eb55ef8b.png)

this still shows the default settings not the missions settings
![missions directories](https://user-images.githubusercontent.com/17562417/55242292-ee240780-520a-11e9-88f9-9605614ac69b.png)

**Desktop (please complete the following information):**
 - OS: Windows 7 Home
 - Version: 5.205.18 [and prior]

**Comment by chummer5a** [comment]
Yeah, that setting's basically a lie. Custom data directories are all global options-based, as data directories are shared between all active characters. I'll need to have a ponder about how to handle that, it was never built for per-character separation.

**Comment by Lochabar** [comment]
This is happening globally - I'm trying to edit the settings for global use across various characters but I do not have a character open. This is the method suggested for #2091 .

**Comment by chummer5a** [comment]
Sorry, per-character and per-setting sort of run together in my head. We store customdatadirectories essentially as a combination of global option and per-setting option in order to accommodate sharing a given setting file and having Chummer throw a warning if the setting file is not present. It's not exactly what you'd expect given other behaviour, obviously. Will probably need some kind of cache invalidation on character switch if the currently cached option is different to what the current character uses.

**Comment by Lochabar** [comment]
Again per-character is not this problem.

I am trying to edit the GLOBAL settings file to be shared.

**Comment by jacx7** [comment]
In Chummer - as I understand it - 'Optional Rules' are always Global. If active, they are used in every character. BUT in the "Character Options" you are able to add/change rules by activating Sourcebooks for your specific Settings File. You can have **Setting specific custom rules** by adding Custom Sourcebooks.

*** I DID NOT TEST THE FOLLOWING FOR YOUR SPECIFIC CASE ***

1. Create a Custom Sourcebook in the '_SR Missions - Neo - Tokyo_' Folder.
```
<?xml version="1.0" encoding="utf-8"?>
<chummer xmlns="" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://www.w3.org/2001/XMLSchema books.xsd">
  <books>
    <book>
      <id>c109b5c5-12f8-4042-b917-14e9281306c7</id>
      <name>CSTM_Mission_Settings_Neo_Tokyo</name>
      <code>CSTM</code>
    </book>
  </books>
</chummer>
```
2. Edit elements of Files in your '_SR Missions - Neo - Tokyo_' Folder so that they refer to your custom Sourcebook. Important lines are:
```
            <page>1</page>
            <source>CSTM</source>
```

Example in context:
```
<?xml version="1.0" encoding="utf-8"?>
<chummer
    xmlns=""
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.w3.org/2001/XMLSchema vehicles.xsd">
    <categories></categories>
    <vehicles>
        <vehicle>
            <id>b81ff748-a6a8-4ee4-a88f-0f554c51beff</id>
            <name>SCMC Darktooth (Custom)</name>
            <page>1</page>
            <source>CSTM</source>
            <accel>2</accel>
            <armor>8</armor>
            <avail>0</avail>
            <body>10</body>
            <category>Cars</category>
            <cost>30000</cost>
            <handling>5/4</handling>
            <pilot>3</pilot>
            <sensor>3</sensor>
            <speed>5</speed>
            <gears>
                <gear>
                    <name>Sensor Array</name>
                    <rating>3</rating>
                    <maxrating>7</maxrating>
                </gear>
            </gears>
            <seats>4</seats>
        </vehicle>
    </vehicles>
    <mods></mods>
</chummer>
```
3. After restarting Chummer: Options > 'Character Options' activate Sourcebook '_CSTM_Mission_Settings_Neo_Tokyo_' for the currently active Ruleset (e.g. Mission Settings).

4. It will be gobaly active but **only used in the specific Settings** file / a character using these Settings.

Hope I got your Issue right - and could help.

**Comment by Lochabar** [comment]
Sorry but no.
As I keep saying I do NOT have a character open. I'm not editing character options.
The problem is with editing the Global files WITHOUT a character being open.

**Comment by chummer5a** [comment]
Like I said, custom data directories are a global option. They're stored in the registry, not the settings file. The only place in which the CharacterOptions.CustomDataDirectoryNames list is used is when loading a character; if any of those rules are missing from the global options on load, it throws a warning. That's it. I'll have to rebuild it to support the selection of folders per characteroption settings file.

**Comment by Lochabar** [comment]
Then what file am I editing in the screenshot above?

I have two files "default" and "missions". When I CREATE a new character I select one of the setting files and this is saved in the character file - with the appropriate set of Custom Data Dictionaries set for the character.

I am trying to modify these global files (no character file open) - default.xml or missions.xml that are present in the "settings" folder of the Chummer5a root folder. This settings folder and default.xml are created when a release of Chummer is used for the first time and do not already exist.

I searched the windows registry. There are a lot more entries in there than there used to be - which goes against the original design of Chummer. However the settings present are the Character Roster, the list of books and the list of active Custom Data Directories. The rest of the options, such as "Life Modules Available" or "Create backup of characters before move them to", are NOT present in the registry - they live in the settings file.

So can we be clear, even though I mentioned the Custom Data Dictionary when I created this issue, that's not the specific problem - it's editing the setting file for other values, not just the custom dictionaries.
